applications/openshift/general/resource_requests_quota/rule.yml

```diff
@@ -0,0 +1,47 @@
+prodtype: ocp4
+
+title: 'Ensure workloads use resource requests and limits'
+
+description: |-
+  
+  There are two ways to enable resource requests and limits. To create either:
+
+  A multi-project quota, defined by a ClusterResourceQuota object, allows quotas
+  to be shared across multiple projects. Resources used in each selected project
+  are aggregated and that aggregate is used to limit resources across all the 
+  selected projects.
+
+  A resource quota, defined by a ResourceQuota object, provides constraints that
+  limit aggregate resource consumption per project. It can limit the quantity of
+  objects that can be created in a project by type, as well as the total amount
+  of compute resources and storage that might be consumed by resources in that project.
+
+  We want to make sure either a ClusterResourceQuota is used in a cluster or a ResourceQuota
+  is used per namespaces.
+
+  <p>
+  To configure ClusterResourceQuota, follow the directions in
+  {{{ weblink(link="https://docs.openshift.com/container-platform/4.9/applications/quotas/quotas-setting-across-multiple-projects.html",
+              text="the documentation") }}}
+  </p>
+
+  <p>
+  To configure ResourceQuota Per Project, follow the directions in
+  {{{ weblink(link="https://docs.openshift.com/container-platform/4.9/applications/quotas/quotas-setting-per-project.html",
+              text="the documentation") }}}
+  </p>
+
+
+rationale: |-
+  Resource quotas provide constraints that limit aggregate resource consumption 
+  per project. This helps prevent resource starvation. When deploying your 
+  application, it is important to tune based on memory and CPU consumption, 
+  allocating enough resources for the application to function properly.
+
+severity: medium
+
+identifiers:
+  cce@ocp4: CCE-90588-5
+
+references:
+  nist: SC-5,SC-5(1),SC-5(2)
```

Comment:
You need to include the warning that tells the compliance operator that it needs to download the resources. Else it won't do it and it'll fail.

Reply:
But this one is extending the other two rules? Do I need to include the warning for both of two rules, `resource_requests_quota_cluster`, `resource_requests_quota_per_project`

Reply:
that's right. The Compliance Operator is not that smart :) So it will only read the warning from the rule that's being used in the data stream. Because of this, the `resource_requests_quota` needs to contain all of the resource inclusions you use.

We could fix this in CO... but it would require further parsing the DS to detect if you're importing other rules in the DS from a single rule...

Reply:
Ok, that's really helpful. thanks for letting me know that! I think adding a warning will fix the problem!